src/app/InteractionModelEngine.cpp

```diff
@@ -387,6 +387,18 @@ void InteractionModelEngine::ShutdownAllSubscriptions()
     ShutdownMatchingSubscriptions();
 }
 
+void InteractionModelEngine::ShutdownAllSubscriptionHandlers()
```

Comment:
I think it would be beneficial to add the associated unit-tests for this function.

Reply:
yes, working on it